### DISCUSSION TITLE
Human-readable string generator for RRULEs

### DISCUSSION THREAD

**Discussion by kriegfrj** [ical4j/ical4j#557] [Ideas]
**Is your feature request related to a problem? Please describe.**

I'd like to be able to describe recurrence rules in a concise, human-readable form.

**Describe the solution you'd like**
A class/method that I can call, which takes a recurrence rule as a parameter, and produces a human-readable string that describes the RRULE. Eg, "Every Monday", "1st Tuesday of every month", "Every day in January of every year", etc.

I recognising that RRules are complex, and that for this reason a concise, human-readable string may not always be easy to produce. Thus, I would settle for a simple implementation that would work for a common subset of RRULE types, and if the RRULE is more complex than what it can handle it can either return a string saying "Sorry, too hard!" (or words to that effect), or else it can throw an exception and let the caller decide how to represent the RRULEs that are too complex. The implementation could be incrementally improved over time.

**Describe alternatives you've considered**
Rolling my own.

**Comment by benfortuna** [comment]
@kriegfrj  I thought this may better suited as a discussion, hope you don't mind. :-)

I think what you are describing probably won't be added directly to the main ical4j library, as this is primarily focused on implementing the RFC specifications (e.g. rfc5545, etc.). That said, I have been thinking about how to extend ical4j to make it easier to create calendar components and properties for common use-cases.

It is still very early days, but I've started working on the ical4j-templates library to address something like this:

https://github.com/ical4j/ical4j-templates

The plan is to extend specific components and properties with a fluent API that makes it easier to create things like meetings, anniversaries, etc.

I could imagine the use-case you described also being potentially address via extending the `RRULE` property to support natural language parsing, or perhaps initially just a more convenient fluent API, e.g.

```
RRule rrule = new FluentRecurrence().every(MONDAY).except(/* some weekday */);

or..

RRule rrule = new FluentRecurrence().first(TUESDAY).until(/* some month/year */);

```

If you have any more suggestions feel free to add to the ical4j-templates issues/disucssions. I will aim to release some initial implementation in the next few weeks.
